src/orchard/info.clj

```diff
@@ -1,13 +1,155 @@
+      (and (not sym) ns)
+      (assoc :unqualified-sym ns)
+
+      (and sym (not (qualified-symbol? sym)) (or ns context-ns))
+      (assoc :qualified-sym (qualify-sym (or ns context-ns) sym)))))
+
+(defn clj-meta
+  {:added "0.6.0"}
+  [{:keys [dialect ns sym computed-ns unqualified-sym]}]
+  {:pre [(= dialect :clj)]}
+  (let [ns (or ns computed-ns)]
+    (or
+     ;; it's a special (special-symbol?)
+     (m/special-sym-meta sym)
+     ;; it's an unqualified sym for an aliased var
+     (some-> ns (m/resolve-var unqualified-sym) (m/var-meta))
+     ;; it's a var
+     (some-> ns (m/resolve-var sym) (m/var-meta))
+     ;; sym is an alias for another ns
+     (some-> ns (m/resolve-aliases) (get sym) (m/ns-meta))
+     ;; We use :unqualified-sym *exclusively* here because because our :ns is
+     ;; too ambiguous.
+     ;;
+     ;; Observe the incorrect behavior (should return nil, there is a test):
+     ;;
+     ;;   (info '{:ns clojure.core :sym non-existing}) ;;=> {:author "Rich Hickey" :ns clojure.core ...}
+     ;;
+     (some-> (find-ns unqualified-sym) (m/ns-meta))
+     ;; it's a Java class/member symbol...or nil
+     (some-> ns (java/resolve-symbol sym)))))
+
+(defn cljs-meta
+  {:added "0.6.0"}
+  [{:keys [dialect ns sym env context-ns unqualified-sym]}]
+  {:pre [(= dialect :cljs)]}
+  (let [context-ns (or context-ns ns)]
+    (or
+     ;; a special symbol - always use :unqualified-sym
+     (some-> (cljs-ana/special-meta env unqualified-sym)
+             (cljs-meta/normalize-var-meta))
+     ;; an NS
+     (some->> (cljs-ana/find-ns env sym)
+              (normalize-ns-meta))
+     ;; ns alias
+     (some->> (cljs-ana/ns-alias env sym context-ns)
+              (cljs-ana/find-ns env)
+              (normalize-ns-meta))
+     ;; macro ns
+     (some->> (find-ns unqualified-sym)
+              (cljs-meta/normalize-macro-ns env))
+
+     ;; macro ns alias
+     (some->> (cljs-meta/aliased-macro-var env sym context-ns)
+              (cljs-meta/normalize-macro-ns env))
+     ;; referred var
+     (some->> (get (cljs-ana/referred-vars env context-ns) sym)
+              (cljs-ana/find-symbol-meta env)
+              (cljs-meta/normalize-var-meta))
+     ;; referred macro
+     (some->> (cljs-meta/referred-macro-meta env sym context-ns)
+              (cljs-meta/normalize-macro-meta))
+     ;; scoped var
+     (some->> (cljs-meta/scoped-var-meta env sym context-ns)
+              (cljs-meta/normalize-var-meta))
+     ;; scoped macro
+     (some->> (cljs-meta/scoped-macro-meta env sym context-ns)
+              (cljs-meta/normalize-macro-meta))
+     ;; var in cljs.core
+     (some->> (get (cljs-ana/core-vars env context-ns) sym)
+              (cljs-ana/var-meta)
+              (cljs-meta/normalize-var-meta))
+     ;; macro in cljs.core
+     (some->> (cljs-meta/scoped-macro-meta env sym 'cljs.core)
+              (cljs-meta/normalize-macro-meta)))))
+
+(defn info
+  [params]
```

Comment:
Switching to a map here is a breaking change. Probably this function should have two signatures to keep backward compatibility (one taking a symbol and ns, and the other taking the map). Not to mention this key function should have a docstring the size of a novel. :D

Reply:
Maybe even a third signature - sym, ns and dialect.

Reply:
How many `orchard` users do we have? 

I kind of think this is a breaking change worth having but if you think it's too much I can change it of course

Reply:
> How many orchard users do we have?

No idea. Probably not many, but I don't think it makes sense to break the API if you can simply add a couple of extra arities to the function.

Reply:
Ok will change it